internal/app/subsystems/api/http/promise.go

```diff
@@ -121,7 +121,7 @@ func (s *server) completePromise(c *gin.Context) {
 		err  error
 	)
 
-	if !body.State.In(promise.Resolved | promise.Rejected | promise.Canceled) {
+	if !body.State.In(promise.Resolved | promise.Rejected | promise.Canceled | promise.Timedout) {
```

Comment:
```suggestion
	if !body.State.In(promise.Resolved | promise.Rejected | promise.Canceled) {
```

A promise can only timeout when tick > promise.Timeout, you cannot time out a promise on request.

Reply:
Thanks, I did the changes accordingly.